modules/account-lib/test/unit/coin/cspr/transactionBuilder/transferBuilder.ts

```diff
@@ -2,91 +2,146 @@ import should from 'should';
+        should.exist((tx as Transaction).casperTx.header.gasPrice, "There is no gasPrice");
+        //TODO (STLX-793) : Check the following fields. To, Amount, gasLimit. 
+      });
 
-        it('a signed transfer transaction from serilaized', async () => {
-          // TODO
-        });
+      it('a transfer transaction with amount 0', async () => {
+        const builder = initTxBuilder();
+        builder.amount('0');
+        builder.sign({ key: testData.ACCOUNT_1.privateKey });
+        const tx = await builder.build();
+        const txJson = JSON.parse(tx.toJson());
+
+        should.exist((tx as Transaction).casperTx.approvals, "There are no approvals");
+        should.deepEqual((tx as Transaction).casperTx.approvals.length, 1, "Error in the number of signatures");
+        should.deepEqual((tx as Transaction).casperTx.approvals[0].signer, testData.SECP256K1_PREFIX + testData.ACCOUNT_1.publicKey, "Error in the signature");
+        should.exist((tx as Transaction).casperTx.hash, "There is no hash");
+        should.exist(txJson.from, "There is no from");
+        should.deepEqual(txJson.from, testData.ACCOUNT_1.publicKey, "The recipient does not match");
+        should.exist((tx as Transaction).casperTx.header.gasPrice, "There is no gasPrice");
+        //TODO (STLX-793) : Check the following fields. To, Amount, gasLimit. 
+      });
 
-        it('an offline multisig transfer transaction', async () => {
-          // TODO
-        });
+      it('a non signed transfer transaction', async () => {
+        const builder = initTxBuilder();
+        const tx = await builder.build();
+        const txJson = JSON.parse(tx.toJson());
+
+        should.deepEqual((tx as Transaction).casperTx.approvals.length, 0, "Error in the number of signatures");
+        should.exist((tx as Transaction).casperTx.hash), "There is no hash";
+        should.exist(txJson.from, "There is no from");
+        should.deepEqual(txJson.from, testData.ACCOUNT_1.publicKey, "The recipient does not match");
+        should.exist((tx as Transaction).casperTx.header.gasPrice, "There is no gasPrice");
+        //TODO (STLX-793) : Check the following fields. To, Amount, gasLimit. 
       });
     });
 
-    describe('should fail', () => {
-      it('a transfer transaction with an invalid key', () => {
+    describe('serialized transactions', () => {
+      it('a non signed transfer transaction from serialized', async () => {
         // TODO
       });
 
-      it('a transfer transaction with more signatures than allowed', () => {
+      it('a signed transfer transaction from serilaized', async () => {
         // TODO
       });
 
-      it('a transfer transaction with repeated sign', () => {
+      it('an offline multisig transfer transaction', async () => {
         // TODO
       });
+    });
+  });
 
-      it('a transfer transaction with an invalid destination address', () => {
-        // TODO
-      });
+  describe('should fail', () => {
```

Comment:
Do we have a ticket to add the missing tests yet?

Reply:
Please add a ticket number for implementing missing tests.

Reply:
Thanks for noticing this @CapnMigraine  .
I did realize that some tests were not added on previous commits, so I just add them.